cypress-tests/cypress/e2e/v2/sessions.cy.ts

```diff
@@ -1,245 +1,287 @@
+      const newName = `${sessionLauncherName} edited`;
+
+      // Click on the session launcher to open the properties view
+      cy.visitProjectByName(projectName);
+      cy.getDataCy("sessions-box")
+        .contains("[data-cy=session-launcher-item]", sessionLauncherName)
+        .click();
+
+      // Edit session launcher
+      cy.getDataCy("session-view-menu-edit").click();
+      cy.getDataCy("edit-session-name")
+        .should("have.value", sessionLauncherName)
+        .clear()
+        .type(newName);
+      cy.getDataCy("edit-session-button").click();
+      cy.getDataCy("session-launcher-update-success").should("be.visible");
+      cy.getDataCy("close-cancel-button").click();
+
+      // Verify session launcher is edited
+      cy.visitProjectByName(projectName);
+      cy.getDataCy("sessions-box").contains(
+        "[data-cy=session-launcher-item]",
+        newName,
+      );
+    });
+
+    it("Edit session launcher container image", () => {
+      const newImage = "ubuntu:latest";
+
+      // Click on the session launcher to open the properties view
+      cy.visitProjectByName(projectName);
+      cy.getDataCy("sessions-box")
+        .contains("[data-cy=session-launcher-item]", sessionLauncherName)
+        .click();
+
+      // Click modify session environment button
+      cy.getDataCy("session-view-modify-session-environment-button").click();
+
+      // Clear and update the container image
+      cy.getDataCy("custom-image-input")
+        .should("have.value", sessionImage)
+        .clear()
+        .type(newImage);
+
+      cy.getDataCy("edit-session-button").click();
+      cy.getDataCy("session-launcher-update-success").should("be.visible");
+      cy.getDataCy("close-cancel-button").click();
+
+      // Verify the image was updated in the properties view
+      cy.visitProjectByName(projectName);
+      cy.getDataCy("sessions-box")
+        .contains("[data-cy=session-launcher-item]", sessionLauncherName)
+        .click();
+
+      // Verify the container image is displayed correctly
+      cy.getDataCy("session-view-session-environment-image").should(
+        "contain",
+        newImage,
       );
     });
+
+    it("Edit session launcher resource class", () => {
+      // Click on the session launcher to open the properties view
+      cy.visitProjectByName(projectName);
+      cy.getDataCy("sessions-box")
+        .contains("[data-cy=session-launcher-item]", sessionLauncherName)
+        .click();
+
+      // Open edit resource class dialog
+      cy.getDataCy("session-view-title").should("be.visible");
+      cy.getDataCy("session-view-resource-class-edit-button").click();
+
+      // Wait for the dialog to open
+      cy.getDataCy("set-resource-class-header").should("be.visible");
+
+      // Open the resource class dropdown
+      cy.get("#addSessionResourceClass").click();
+
+      // Select the "medium" resource class option
+      cy.contains("medium").filter(":visible").click();
```

Comment:
In lines 228-231, I wasn't sure how to handle the selection of a resource class in the dropdown menu since it's not a standard one.